Feature/gdb

BaranovMykola opened the pull request:
This classes are core of using gdb in Kuzya. This pull request has class Gdb wich provides basics methods such as step out\into\over, reading variables, etc. Also it has class Variable and Breakpoint. This pull request doesn't make any influence to kuzya, because this classes are not using yet.

BaranovMykola committed:
Added thesis about GDB integration

BaranovMykola committed:
Moved thesis to doc/conj/fosslviv2017 folder

BaranovMykola committed:
Deleted thesis from root directory

BaranovMykola committed:
Added thesis in .odt format

BaranovMykola committed:
Changed thesis: removed part about kuzya's delegate, added general describe of gdb struct in kuzya

BaranovMykola committed:
Added class Gdb wich provides using GDB debugger. Added attendant classes Variable and Breakpoint

BaranovMykola committed:
Added method stepContinue - continue normal executing of the programm

OlyaOlga commented on src/breakpoint.cpp:
```diff
@@ -0,0 +1,69 @@
+    /*
+        $"1       breakpoint     keep y   0x0040149e in main() at main.cpp:40\n"$
+        $"2       breakpoint     del y   0x004014fc in main() at main.cpp:41\n"$
+    */
```
Is this comment necessary?

BaranovMykola replied:
This comment shows you format of input string

OlyaOlga commented on src/variable.cpp:
```diff
@@ -0,0 +1,184 @@
+
+Variable::Variable()
+{
+
```
Delete this line, please.

OlyaOlga commented on src/variable.cpp:
```diff
@@ -0,0 +1,184 @@
+
+QStringList Variable::getSubVariables() const
+{  // returns QStringList consists from nested variable's names (only first level)
+   //  {digit = {real = 3, imaginary = 4}}
```
Seems like this comment doesn't match the context of the code.

OlyaOlga commented on src/variable.cpp:
```diff
@@ -0,0 +1,184 @@
+//        nestedType = this->readNestedStruct(nestedType);
+//        res << nestedType;
+//    }
+//    while(!nestedType.isEmpty());
```
Delete unused code, please.

OlyaOlga commented on src/variable.cpp:
```diff
@@ -0,0 +1,184 @@
+            capture = true;
+            if(key.isEmpty() && value.isEmpty())
+            {
+                return res;
```
Why do you need a return statement here?

OlyaOlga commented on src/variable.cpp:
```diff
@@ -0,0 +1,184 @@
+            {
+                value[0] = ' ';
+                value[1] = ' ';
+//                value.append('}');
```
Is this comment necessary here?

BaranovMykola committed:
Removed some extra comments, moved return statement to the end of Variable::readNestedStruct method

BaranovMykola committed:
Added method setGdbPath which can set path to gdb, added checking condition of running gdb before writing, added method stopExecuting wich kills process of target programm, added exceptions

BaranovMykola committed:
Removed commented code, fixed grammar error in some comment

BaranovMykola committed:
Removed warnings produced by regular expression at line 214 and line 183

BaranovMykola committed:
Removed thesis about GDB which left after creating new branch from Thesis/39-thesis-about-gdb

BaranovMykola committed:
Changed using exception to emitting signalErrorOccured in Gdb::write method

BaranovMykola committed:
Added icon 'run in debug mode' to main menu/build and toolbar. Processed visible icon

BaranovMykola committed:
Added simple static configuration Gdb class in Kuzya constructor

BaranovMykola committed:
Run programm under GDB

BaranovMykola committed:
Included Gdb class to Kuzya

BaranovMykola committed:
Changed signature of wirte() in Gdb class which may cause test fails

BaranovMykola committed:
Changed signature of write() in Gdb.cpp file

BaranovMykola committed:
Removed throwing exception which may couse test fails

BaranovMykola committed:
Removed some stl exception which may couse test fails

BaranovMykola committed:
Try to restore losted commits

BaranovMykola committed:
commited merge

BaranovMykola committed:
Used std::invalid_argument exception instend of std::exception

BaranovMykola committed:
Using std::domain_error while user try to open gdb via incorrect path

BaranovMykola committed:
Added getter of programPath to Compiler class. Added slotDebugRun to
kuzya which compile and run programm in debug mode. Added condition to
check out if Gdb exist. If not - catch exception and show error to
notificationList. Added icon to Kuya form "Run in Debug Mode" and
relevant item in menu "build". Added icon "start debug".

BaranovMykola committed:
Deleted extra form with classes

BaranovMykola committed:
Removed extra icon, removed qDebug from Kuzya constructor

BaranovMykola committed:
Added breakpoint.png icon. Defined breapoint marker in text editor,
setted new byte mask for left margin. Breakpoint can be added by Alt
+Left Mouse Button (only graphic)

BaranovMykola committed:
Splitted processing click in error point and setting breakpoints in
slotMarginClicked. Added opportunity to delete existed breakpoint (only
icon).

BaranovMykola committed:
Fixed losted warning mask. Moved mask contant to enum
marginMarkerMaskType, used that instend of numeric constant

BaranovMykola committed:
Defined each marker by numeric constant, because dynamic defining may
cause different numeric. MarginMargerMasks uses it's numeric to define
position at margin. Dynamic defining may cause random position of
markers.

BaranovMykola committed:
Fixed removing all breakpoints while clearing notification list with all
notification markers (warning, error, current).

BaranovMykola committed:
Changed warning to error constant again

BaranovMykola committed:
Removed extra slot with it's connection

BaranovMykola committed:
Merge pull request #54 from PLLUG/gdb/43-run-in-debug-mode

Gdb/53 - introduce Gdb class to Kuzya, 43 - run in debug mode

BaranovMykola committed:
Merge pull request #55 from PLLUG/gdb/51-add-breakpoints

Gdb/51 - add breakpoints

OlyaOlga committed:
added tabWidget, moved notification list inside tabWidget

OlyaOlga committed:
added tab 'Debug' in tabWidget, added mTreeViewWidget(QTreeWidget), inserted mTreeViewWidget and debugButtons(QToolBar) widgets into tab 'Debug'

OlyaOlga committed:
removed qDebug output, added comment

OlyaOlga committed:
removed unnecessary code, renamed  mTreeViewWidget to mWatchLocalsWidget and tabWidget to mOutputTabWidget

OlyaOlga committed:
removed unnecessary piece of code

OlyaOlga committed:
Merge pull request #59 from PLLUG/gdb/50-output-panel

Gdb/50 output panel
